### DISCUSSION TITLE
Unable to proceed with 3D secure payment on Stripe testing (able to click on iFrame)

### DISCUSSION THREAD

**Discussion by TheHoneyBadger13** [cypress-io/cypress#26118] [Questions and Help]
I am trying to test a stripe web app (not sure what the environment is, but looks like a test one). I can proceed with non 3d secure payment but on 3d payments, it is stuck on processing. Not sure if the problem is with handling the iframe but I seem to be able to click the button. I also made a repo in case someone wants to try and replicate it, not sure which code I should provide cause I am not sure if it is on the handling of elements or API related
github repo: https://github.com/TheHoneyBadger13/stripe-checkout-test

![image](https://user-images.githubusercontent.com/38536130/225227428-caa71609-5d5f-4d0c-a457-4c1e809b1b18.png)
After pressing the Complete, it gets stuck on processing instead of showing a success screen/animation
![image](https://user-images.githubusercontent.com/38536130/225227340-249edcd8-63c1-4039-b04e-6b05eb1c622c.png)

**Comment by portyu9** [comment]
Hi @TheHoneyBadger13 

If the Stripe 3D Secure challenge is running inside a **cross-origin iframe**, this is a Cypress/browser same-origin limitation rather than a selector problem.

Cypress's current documentation specifically calls out Stripe/Braintree credit-card frames as examples of cross-origin iframes that Cypress cannot directly automate or communicate with.

`cy.origin()` does **not** solve this case. It supports a secondary **top-level origin**, but it cannot run commands inside an `<iframe>`.

That means I would not build the main payment test around clicking controls inside Stripe's 3DS challenge iframe.

A more reliable test strategy is:

1. Test that your application creates the payment correctly.
2. Use Stripe's test environment/test payment states to exercise the required 3DS outcome.
3. Test your own application's handling of success/failure/cancellation deterministically, stubbing your backend/payment state or webhook where appropriate.
4. Keep a smaller provider-integration smoke test if you need end-to-end coverage of Stripe itself.

If your particular Stripe flow redirects the **whole browser page** to another origin instead of embedding the challenge in an iframe, that is different: current Cypress can handle top-level cross-origin interaction with `cy.origin()`.

The repository linked in the question is no longer accessible, so I cannot verify which exact Stripe integration or iframe helper was being used there.

Please mark this answer as accepted if it helped, thank you.
